### PR TITLE
Restore Laravel 6 behavior to public visibility for local driver

### PR DESCRIPTION
**Pull request by mjauvin**
Fixes https://github.com/wintercms/winter/issues/503

@LukeTowers @bennothommo @jaxwilko

Here's my preferred solution to this.

### CODE REVIEW

File: src/Filesystem/FilesystemManager.php [hunk 1]
```diff
@@ -22,4 +22,16 @@ public function identify($disk)
         }
         return $configName;
     }
+
+    protected function resolve($name, $config = null)
+    {
+        if (is_null($config)) {
+            $config = $this->getConfig($name);
+        }
+        if ($name === 'local') {
```
Comment (LukeTowers) [inline src/Filesystem/FilesystemManager.php]: Shouldn't this be the following instead?
```php
if ($config['driver'] === 'local' && empty($config['visibility']))
```

**Comment by LukeTowers** [comment]
@driesvints if you have a moment do you recall or have a link to the reasoning behind changing the default from public to private for the local Flysystem driver in Laravel 9?

**Comment by driesvints** [comment]
@LukeTowers that got changed in Flysystem v3 so you'll have to ask them. Afaik it's a more safe and sensible default. You should always use the same user to handle files.

**Comment by LukeTowers** [comment]
@driesvints I agree in theory that it's more safe default, the PHP user can access the files but the problem comes when the webserver user wants to fetch them to respond to requests, which is the default used by a few providers, ServerPilot & Cleavr for example. 

Any recommendations for how to handle that use case?

**Comment by driesvints** [comment]
Not really sorry..

File: src/Filesystem/FilesystemManager.php (hunk 1 above)
Comment (mjauvin) [reply src/Filesystem/FilesystemManager.php]: That works too

**Comment by LukeTowers** [comment]
Thanks anyways :)

File: src/Filesystem/FilesystemManager.php (hunk 1 above)
Comment (LukeTowers) [inline src/Filesystem/FilesystemManager.php]:
```suggestion
        if ($config['driver'] === 'local' && empty($config['visibility']))
```

File: src/Filesystem/FilesystemManager.php [hunk 2]
```diff
@@ -22,4 +22,16 @@ public function identify($disk)
         }
         return $configName;
     }
+
```
Comment (LukeTowers) [inline src/Filesystem/FilesystemManager.php:25]:
```suggestion

    /**
     * @inheritDoc
     */
```

File: src/Filesystem/FilesystemManager.php [hunk 3]
```diff
@@ -22,4 +22,19 @@ public function identify($disk)
...
+    /**
+     * @inheritDoc
+     */
+    protected function resolve($name, $config = null)
+    {
+        if (is_null($config)) {
+            $config = $this->getConfig($name);
+        }
+        if ($config['driver'] === 'local' && empty($config['visibility']))
```
Comment (LukeTowers) [inline src/Filesystem/FilesystemManager.php]:
```suggestion
        if ($config['driver'] === 'local' && empty($config['visibility'])) {
```

File: src/Filesystem/FilesystemManager.php [hunk 4]
```diff
@@ -22,4 +22,19 @@ public function identify($disk)
...
+
+    /**
+     * @inheritDoc
+     */
+    protected function resolve($name, $config = null)
+    {
+        if (is_null($config)) {
+            $config = $this->getConfig($name);
+        }
+        if ($config['driver'] === 'local' && empty($config['visibility'])) {
```
Comment (LukeTowers) [inline src/Filesystem/FilesystemManager.php:37]:
```suggestion

        // Default local drivers to public visibility for backwards compatibility
        // see https://github.com/wintercms/winter/issues/503
        if ($config['driver'] === 'local' && empty($config['visibility'])) {
```

**Pull request merged**